doc/nrf/gs_modifying.rst

```diff
@@ -154,3 +154,76 @@ To specify CMake options, click :guilabel:`Tools` > :guilabel:`Options`, select
+
+.. build_types_overview_start
+
+Build types enable you to use different sets of configuration options for each board.
+You can create several build type ``.conf`` files per board and select one of them when building the application.
+This means that you do not have to use one :file:`prj.conf` file for your project and modify it each time to fit your needs.
+
+.. build_types_overview_end
+
+.. note::
+    Creating build types and selecting them is optional.
+    This is a feature specific to the :ref:`application development in |NCS| <ncs-app-dev>`.
+
+.. _gs_modifying_build_types_creating:
+
+Creating build type files
+=========================
+
+To create custom build type files for your application instead of using a single :file:`prj.conf` file, complete the following steps:
+
+1. During :ref:`application development <zephyr:application>`, follow the procedure for creating the application until after the step where you create the :file:`CMakeLists.txt` file.
+#. In the :file:`CMakeLists.txt` file, define the file name pattern for configuration files.
+   For example::
+
+    set(CONF_FILE "app_${CMAKE_BUILD_TYPE}.conf")
+
+   In this define, ``CMAKE_BUILD_TYPE`` will be used for selecting the build type.
+#. Optionally, include an if statement that checks for the presence of the selected build type configuration files.
+   For an example, see :file:`applications/nrf_desktop/CMakeLists.txt`.
+#. Continue the application creation procedure by setting the Kconfig configuration options.
+#. Save the :file:`.conf` file in the application directory with a name that matches the file name pattern defined in CMakeLists.
+   For example, :file:`app_ZRelease.conf`.
+   In this file name, ``ZRelease`` is the build type name.
+
+You can now select build types in SES or from command line.
+
+Selecting a build type in SES
+=============================
+
+.. build_types_selection_ses_start
+
+To select the build type in SEGGER Embedded Studio:
+
+1. Go to :guilabel:`Tools` -> :guilabel:`Options...` -> :guilabel:`nRF Connect`.
+#. Set ``Additional CMake Options`` to ``-DCMAKE_BUILD_TYPE=selected_build_type``.
+   For example, for ``ZRelease`` set the following value: ``-DCMAKE_BUILD_TYPE=ZRelease``.
+#. Reload the project.
+
+The changes will be applied after reloading.
+
+.. build_types_selection_ses_end
+
+Selecting a build type from command line
+========================================
+
+.. build_types_selection_cmd_start
+
+To select the build type when building the application from command line, specify the build type by adding the ``-- -DCMAKE_BUILD_TYPE=selected_build_type`` to the ``west build`` command.
+
+For example, you can build the ``ZRelease`` firmware for the PCA20041 board by running the following command in the project directory:
+
+.. code-block:: console
+
+   west build -b nrf52840_pca20041 -d build_pca20041 -- -DCMAKE_BUILD_TYPE=ZRelease
+
+The ``build_pca20041`` parameter specifies the output directory for the build files.
```

Comment:
Old board naming convention, I think we should use new naming convention.
You could also refer to universal board, not dedicated for nrf_desktop project:
for example nrf52840dk_nrf52840 (that is DK- pca10056).

Reply:
I think it was already there. We have not fixed the doc regarding the board names. Let's leave it for a separate PR.

Reply:
Yes, this is a different and unrelated issue.